Python tests failing

In of e176d8ae5a267148387ac9500a051ce5de253014 (current `master` at the time of writing) 7 tests in the `test_cuda_simulation.py` of python tests are failing. 
They are all `test_argparse_*` tests. I do not know how long they have been broken.

Under linux, with CUDA 11.7, python 3.9.5 gcc `-DCMAKE_BUILD_TYPE=Release -DSEATBELTS=ON -DCUDA_ARCH=86 -DBUILD_SWIG_PYTHON=ON` build:

```text
python3 -m pytest ../tests/swig/python/gpu/test_cuda_simulation.py 
================================================================ test session starts ================================================================
platform linux -- Python 3.9.5, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/ptheywood/code/flamegpu/FLAMEGPU2
collected 19 items                                                                                                                                  

../tests/swig/python/gpu/test_cuda_simulation.py ..FFFsFFFF.........                                                                          [100%]

===================================================================== FAILURES ======================================================================
_____________________________________________________ TestSimulation.test_argparse_device_long ______________________________________________________

self = <test_cuda_simulation.TestSimulation testMethod=test_argparse_device_long>

    def test_argparse_device_long(self):
        m = pyflamegpu.ModelDescription("test_argparse_device_long")
        c = pyflamegpu.CUDASimulation(m)
        argv = [ "prog.exe", "--device", "1200" ]
        assert c.CUDAConfig().device_id == 0
        # Setting an invalid device ID is the only safe way to do this without making internal methods accessible
        # As can set to a valid device, we haven't build code for
        with pytest.raises(pyflamegpu.FLAMEGPURuntimeException) as e:  # exception::InvalidCUDAdevice exception
            c.initialise(argv)
        assert e.value.type() == "InvalidCUDAdevice"
        assert c.CUDAConfig().device_id == 1200
        # Blank init resets value to default
        argv = []
>       c.initialise(argv)

../tests/swig/python/gpu/test_cuda_simulation.py:134: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd02a6b7270> >, argc = []

    def initialise(self, argc):
>       return _pyflamegpu.Simulation_initialise(self, argc)
E       pyflamegpu.pyflamegpu.FLAMEGPURuntimeException: (InvalidCUDAdevice) /home/ptheywood/code/flamegpu/FLAMEGPU2/src/flamegpu/gpu/CUDASimulation.cu(1457): Error setting CUDA device to '1200', only 1 available!

lib/Release/python/venv/lib/python3.9/site-packages/pyflamegpu/pyflamegpu.py:6966: FLAMEGPURuntimeException
_____________________________________________________ TestSimulation.test_argparse_device_short _____________________________________________________

self = <test_cuda_simulation.TestSimulation testMethod=test_argparse_device_short>

    def test_argparse_device_short(self):
        m = pyflamegpu.ModelDescription("test_argparse_device_short")
        c = pyflamegpu.CUDASimulation(m)
        argv = [ "prog.exe", "-d", "1200" ]
        assert c.CUDAConfig().device_id == 0
        # Setting an invalid device ID is the only safe way to do this without making internal methods accessible
        # As can set to a valid device, we haven't build code for
        with pytest.raises(pyflamegpu.FLAMEGPURuntimeException) as e:  # exception::InvalidCUDAdevice exception
            c.initialise(argv)
        assert e.value.type() == "InvalidCUDAdevice"
        assert c.CUDAConfig().device_id == 1200
        # Blank init resets value to default
        argv = []
>       c.initialise(argv)

../tests/swig/python/gpu/test_cuda_simulation.py:150: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd022e95b10> >, argc = []

    def initialise(self, argc):
>       return _pyflamegpu.Simulation_initialise(self, argc)
E       pyflamegpu.pyflamegpu.FLAMEGPURuntimeException: (InvalidCUDAdevice) /home/ptheywood/code/flamegpu/FLAMEGPU2/src/flamegpu/gpu/CUDASimulation.cu(1457): Error setting CUDA device to '1200', only 1 available!

lib/Release/python/venv/lib/python3.9/site-packages/pyflamegpu/pyflamegpu.py:6966: FLAMEGPURuntimeException
____________________________________________________ TestSimulation.test_argparse_inputfile_long ____________________________________________________

self = <test_cuda_simulation.TestSimulation testMethod=test_argparse_inputfile_long>

    def test_argparse_inputfile_long(self):
        m = pyflamegpu.ModelDescription("test_argparse_inputfile_long")
        c = pyflamegpu.CUDASimulation(m)
        argv = [ "prog.exe", "--in", "test" ]
        assert c.SimulationConfig().input_file == ""
        with pytest.raises(pyflamegpu.FLAMEGPURuntimeException) as e:  # exception::UnsupportedFileType exception
            c.initialise(argv)
        assert e.value.type() == "UnsupportedFileType"
        assert c.SimulationConfig().input_file == argv[2]
        # Blank init resets value to default
        argv = []
>       c.initialise(argv)

../tests/swig/python/gpu/test_cuda_simulation.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd0228c5960> >, argc = []

    def initialise(self, argc):
>       return _pyflamegpu.Simulation_initialise(self, argc)
E       pyflamegpu.pyflamegpu.FLAMEGPURuntimeException: (UnsupportedFileType) /home/ptheywood/code/flamegpu/FLAMEGPU2/include/flamegpu/io/StateReaderFactory.h(52): File 'test' is not a type which can be read by StateReaderFactory::createReader().

lib/Release/python/venv/lib/python3.9/site-packages/pyflamegpu/pyflamegpu.py:6966: FLAMEGPURuntimeException
___________________________________________________ TestSimulation.test_argparse_randomseed_long ____________________________________________________

self = <test_cuda_simulation.TestSimulation testMethod=test_argparse_randomseed_long>

    def test_argparse_randomseed_long(self):
        m = pyflamegpu.ModelDescription("test_argparse_randomseed_long")
        c = pyflamegpu.CUDASimulation(m)
        argv = [ "prog.exe", "--random", "12" ]
        assert c.SimulationConfig().random_seed != 12
        c.initialise(argv)
        assert c.SimulationConfig().random_seed == 12
        # Blank init resets value to default
        argv = []
        c.initialise(argv)
>       assert c.SimulationConfig().random_seed != 12
E       AssertionError: assert 12 != 12
E        +  where 12 = <pyflamegpu.pyflamegpu.Simulation_Config; proxy of <Swig Object of type 'flamegpu::Simulation::Config *' at 0x7fd02a6de9f0> >.random_seed
E        +    where <pyflamegpu.pyflamegpu.Simulation_Config; proxy of <Swig Object of type 'flamegpu::Simulation::Config *' at 0x7fd02a6de9f0> > = <bound method Simulation.SimulationConfig of <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd02a6deab0> >>()
E        +      where <bound method Simulation.SimulationConfig of <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd02a6deab0> >> = <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd02a6deab0> >.SimulationConfig

../tests/swig/python/gpu/test_cuda_simulation.py:107: AssertionError
___________________________________________________ TestSimulation.test_argparse_randomseed_short ___________________________________________________

self = <test_cuda_simulation.TestSimulation testMethod=test_argparse_randomseed_short>

    def test_argparse_randomseed_short(self):
        m = pyflamegpu.ModelDescription("test_argparse_randomseed_short")
        c = pyflamegpu.CUDASimulation(m)
        argv = [ "prog.exe", "-r", "12" ]
        assert c.SimulationConfig().random_seed != 12
        c.initialise(argv)
        assert c.SimulationConfig().random_seed == 12
        # Blank init resets value to default
        argv = []
        c.initialise(argv)
>       assert c.SimulationConfig().random_seed != 12
E       AssertionError: assert 12 != 12
E        +  where 12 = <pyflamegpu.pyflamegpu.Simulation_Config; proxy of <Swig Object of type 'flamegpu::Simulation::Config *' at 0x7fd02296f690> >.random_seed
E        +    where <pyflamegpu.pyflamegpu.Simulation_Config; proxy of <Swig Object of type 'flamegpu::Simulation::Config *' at 0x7fd02296f690> > = <bound method Simulation.SimulationConfig of <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd02296f900> >>()
E        +      where <bound method Simulation.SimulationConfig of <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd02296f900> >> = <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd02296f900> >.SimulationConfig

../tests/swig/python/gpu/test_cuda_simulation.py:119: AssertionError
______________________________________________________ TestSimulation.test_argparse_steps_long ______________________________________________________

self = <test_cuda_simulation.TestSimulation testMethod=test_argparse_steps_long>

    def test_argparse_steps_long(self):
        m = pyflamegpu.ModelDescription("test_argparse_steps_long")
        c = pyflamegpu.CUDASimulation(m)
        argv = [ "prog.exe", "--steps", "12" ]
        assert c.SimulationConfig().steps == 1
        c.initialise(argv)
        assert c.SimulationConfig().steps == 12
        # Blank init resets value to default
        argv = []
        c.initialise(argv)
>       assert c.SimulationConfig().steps == 1
E       AssertionError: assert 12 == 1
E        +  where 12 = <pyflamegpu.pyflamegpu.Simulation_Config; proxy of <Swig Object of type 'flamegpu::Simulation::Config *' at 0x7fd022968a50> >.steps
E        +    where <pyflamegpu.pyflamegpu.Simulation_Config; proxy of <Swig Object of type 'flamegpu::Simulation::Config *' at 0x7fd022968a50> > = <bound method Simulation.SimulationConfig of <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd022968570> >>()
E        +      where <bound method Simulation.SimulationConfig of <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd022968570> >> = <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd022968570> >.SimulationConfig

../tests/swig/python/gpu/test_cuda_simulation.py:83: AssertionError
_____________________________________________________ TestSimulation.test_argparse_steps_short ______________________________________________________

self = <test_cuda_simulation.TestSimulation testMethod=test_argparse_steps_short>

    def test_argparse_steps_short(self):
        m = pyflamegpu.ModelDescription("test_argparse_steps_short")
        c = pyflamegpu.CUDASimulation(m)
        argv = [ "prog.exe", "-s", "12" ]
        assert c.SimulationConfig().steps == 1
        c.initialise(argv)
        assert c.SimulationConfig().steps == 12
        # Blank init resets value to default
        argv = []
        c.initialise(argv)
>       assert c.SimulationConfig().steps == 1
E       AssertionError: assert 12 == 1
E        +  where 12 = <pyflamegpu.pyflamegpu.Simulation_Config; proxy of <Swig Object of type 'flamegpu::Simulation::Config *' at 0x7fd0229cfc00> >.steps
E        +    where <pyflamegpu.pyflamegpu.Simulation_Config; proxy of <Swig Object of type 'flamegpu::Simulation::Config *' at 0x7fd0229cfc00> > = <bound method Simulation.SimulationConfig of <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd0229cf390> >>()
E        +      where <bound method Simulation.SimulationConfig of <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd0229cf390> >> = <pyflamegpu.pyflamegpu.CUDASimulation; proxy of <Swig Object of type 'flamegpu::CUDASimulation *' at 0x7fd0229cf390> >.SimulationConfig

../tests/swig/python/gpu/test_cuda_simulation.py:95: AssertionError
============================================================== short test summary info ==============================================================
FAILED ../tests/swig/python/gpu/test_cuda_simulation.py::TestSimulation::test_argparse_device_long - pyflamegpu.pyflamegpu.FLAMEGPURuntimeExceptio...
FAILED ../tests/swig/python/gpu/test_cuda_simulation.py::TestSimulation::test_argparse_device_short - pyflamegpu.pyflamegpu.FLAMEGPURuntimeExcepti...
FAILED ../tests/swig/python/gpu/test_cuda_simulation.py::TestSimulation::test_argparse_inputfile_long - pyflamegpu.pyflamegpu.FLAMEGPURuntimeExcep...
FAILED ../tests/swig/python/gpu/test_cuda_simulation.py::TestSimulation::test_argparse_randomseed_long - AssertionError: assert 12 != 12
FAILED ../tests/swig/python/gpu/test_cuda_simulation.py::TestSimulation::test_argparse_randomseed_short - AssertionError: assert 12 != 12
FAILED ../tests/swig/python/gpu/test_cuda_simulation.py::TestSimulation::test_argparse_steps_long - AssertionError: assert 12 == 1
FAILED ../tests/swig/python/gpu/test_cuda_simulation.py::TestSimulation::test_argparse_steps_short - AssertionError: assert 12 == 1
====================================================== 7 failed, 11 passed, 1 skipped in 0.57s ======================================================
```

--- a/tests/swig/python/gpu/test_cuda_simulation.py
+++ b/tests/swig/python/gpu/test_cuda_simulation.py
@@ -50,14 +50,32 @@ class TestSimulation(TestCase):
             c.initialise(argv)
         assert e.value.type() == "UnsupportedFileType"
         assert c.SimulationConfig().input_file == argv[2]
-        # Blank init resets value to default
-        argv = []
-        c.initialise(argv)
-        assert c.SimulationConfig().input_file == ""
+        # Blank init does not reset value to default
+        with pytest.raises(pyflamegpu.FLAMEGPURuntimeException) as e:  # exception::UnsupportedFileType exception
+            c.initialise([])
+        assert e.value.type() == "UnsupportedFileType"
+        # assert c.SimulationConfig().input_file == ""
+        assert c.SimulationConfig().input_file == argv[2]
 
-    @pytest.mark.skip(reason="no way of intercepting exit called with c/c++ code from python")
     def test_argparse_inputfile_short(self):
         m = pyflamegpu.ModelDescription("test_argparse_inputfile_short")
+        c = pyflamegpu.CUDASimulation(m)
+        argv = [ "prog.exe", "-i", "test" ]
+        assert c.SimulationConfig().input_file == ""
+        with pytest.raises(pyflamegpu.FLAMEGPURuntimeException) as e:  # exception::UnsupportedFileType exception
+            c.initialise(argv)
+        assert e.value.type() == "UnsupportedFileType"
+        assert c.SimulationConfig().input_file == argv[2]
+        # Blank init does not reset value to default
+        with pytest.raises(pyflamegpu.FLAMEGPURuntimeException) as e:  # exception::UnsupportedFileType exception
+            c.initialise([])
+        assert e.value.type() == "UnsupportedFileType"
+        # assert c.SimulationConfig().input_file == ""
+        assert c.SimulationConfig().input_file == argv[2]
+
+    @pytest.mark.skip(reason="no way of intercepting exit called with c/c++ code from python")
+    def test_argparse_inputfile_short_exit(self):
+        m = pyflamegpu.ModelDescription("test_argparse_inputfile_short_exit")
         c = pyflamegpu.CUDASimulation(m)
         argv = [ "prog.exe", "-i", "I_DO_NOT_EXIST.xml" ]
         assert c.SimulationConfig().input_file == ""
@@ -77,10 +95,9 @@ class TestSimulation(TestCase):
         assert c.SimulationConfig().steps == 1
         c.initialise(argv)
         assert c.SimulationConfig().steps == 12
-        # Blank init resets value to default
-        argv = []
-        c.initialise(argv)
-        assert c.SimulationConfig().steps == 1
+        # Blank does not reset value to default
+        c.initialise([])
+        assert c.SimulationConfig().steps == 12
         
     def test_argparse_steps_short(self):
         m = pyflamegpu.ModelDescription("test_argparse_steps_short")
@@ -89,10 +106,9 @@ class TestSimulation(TestCase):
         assert c.SimulationConfig().steps == 1
         c.initialise(argv)
         assert c.SimulationConfig().steps == 12
-        # Blank init resets value to default
-        argv = []
-        c.initialise(argv)
-        assert c.SimulationConfig().steps == 1
+        # Blank does not reset value to default
+        c.initialise([])
+        assert c.SimulationConfig().steps == 12
         
     def test_argparse_randomseed_long(self):
         m = pyflamegpu.ModelDescription("test_argparse_randomseed_long")
@@ -101,10 +117,9 @@ class TestSimulation(TestCase):
         assert c.SimulationConfig().random_seed != 12
         c.initialise(argv)
         assert c.SimulationConfig().random_seed == 12
-        # Blank init resets value to default
-        argv = []
-        c.initialise(argv)
-        assert c.SimulationConfig().random_seed != 12
+        # Blank does not reset value to default
+        c.initialise([])
+        assert c.SimulationConfig().random_seed == 12
         
     def test_argparse_randomseed_short(self):
         m = pyflamegpu.ModelDescription("test_argparse_randomseed_short")
@@ -113,10 +128,9 @@ class TestSimulation(TestCase):
         assert c.SimulationConfig().random_seed != 12
         c.initialise(argv)
         assert c.SimulationConfig().random_seed == 12
-        # Blank init resets value to default
-        argv = []
-        c.initialise(argv)
-        assert c.SimulationConfig().random_seed != 12
+        # Blank does not reset value to default
+        c.initialise([])
+        assert c.SimulationConfig().random_seed == 12
         
     def test_argparse_device_long(self):
         m = pyflamegpu.ModelDescription("test_argparse_device_long")
@@ -129,10 +143,12 @@ class TestSimulation(TestCase):
             c.initialise(argv)
         assert e.value.type() == "InvalidCUDAdevice"
         assert c.CUDAConfig().device_id == 1200
-        # Blank init resets value to default
+        # Blank init does not reset value to default
         argv = []
-        c.initialise(argv)
-        assert c.CUDAConfig().device_id == 0
+        with pytest.raises(pyflamegpu.FLAMEGPURuntimeException) as e:  # exception::InvalidCUDAdevice exception 
+            c.initialise(argv)
+            assert e.value.type() == "InvalidCUDAdevice"
+        assert c.CUDAConfig().device_id == 1200
         
     def test_argparse_device_short(self):
         m = pyflamegpu.ModelDescription("test_argparse_device_short")
@@ -145,10 +161,12 @@ class TestSimulation(TestCase):
             c.initialise(argv)
         assert e.value.type() == "InvalidCUDAdevice"
         assert c.CUDAConfig().device_id == 1200
-        # Blank init resets value to default
+        # Blank init does not reset value to default
         argv = []
-        c.initialise(argv)
-        assert c.CUDAConfig().device_id == 0
+        with pytest.raises(pyflamegpu.FLAMEGPURuntimeException) as e:  # exception::InvalidCUDAdevice exception 
+            c.initialise(argv)
+            assert e.value.type() == "InvalidCUDAdevice"
+        assert c.CUDAConfig().device_id == 1200
 
 
     SetGetFn = """

